test/support/test_config.h

```diff
@@ -16,6 +16,21 @@
 #ifndef _TEST_config_H
 #define _TEST_config_H
 
+/* Make a nice name for the standard version */
+#ifndef TEST_STD_VER
+#    if __cplusplus >= 202002L || _MSVC_LANG >= 202002L
+#        define TEST_STD_VER 20
+#    elif __cplusplus >= 201703L || _MSVC_LANG >= 201703L
+#        define TEST_STD_VER 17
+#    elif __cplusplus >= 201402L || _MSVC_LANG >= 201402L
+#        define TEST_STD_VER 14
+#    elif __cplusplus >= 201103L || _MSVC_LANG >= 201103L
+#        define TEST_STD_VER 11
+#    elif __cplusplus >= 199711L || _MSVC_LANG >= 199711L
+#        define TEST_STD_VER 3
+#    endif

```

Comment:
I think we can make this an `#else` branch, i.e. assume C++03 if all the above checks fail.

```suggestion
#    else // assume C++03
#        define TEST_STD_VER 3
#    endif

```